c2rust-transpile/src/c_ast/mod.rs

```diff
@@ -536,11 +536,14 @@ impl TypedAstContext {
                 CDeclKind::Function {
                     body: Some(_),
                     is_global: true,
-                    is_inline: false,
+                    is_inline: is_inl,
+                    is_extern: is_ext,
                     ..
                 } => {
-                    to_walk.push(decl_id);
-                    used.insert(decl_id);
+                    if !is_inl || (is_inl && is_ext) {
```

Comment:
Didn't my suggestion work? You shouldn't need a separate `if`, it could just be a guard in the match arm.